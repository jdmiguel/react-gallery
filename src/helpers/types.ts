export type ThumbData = {
  id: number;
  src: string;
  title: string;
}

export type DetailData = {
  id: number;
  src: string;
  title: string;
  views: number;
  downloads: number;
  favorites: number;
  likes: number;
}

export type ImageExtendedData = {
  id: number;
  pageURL: string;
  type: string;
  tags: string;
  previewURL: string;
  previewWidth: number;
  previewHeight: number;
  webformatURL: string;
  webformatWidth: number;
  webformatHeight: number;
  largeImageURL: string;
  fullHDURL: string;
  imageURL: string;
  imageWidth: number;
  imageHeight: number;
  imageSize: number;
  views: number;
  downloads: number;
  favorites: number;
  likes: number;
  comments: number;
  user_id: number;
  user: string;
  userImageURL: string;
}