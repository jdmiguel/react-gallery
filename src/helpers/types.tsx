export type Image = {
  src: string;
}