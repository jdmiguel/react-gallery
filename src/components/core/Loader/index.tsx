import React from 'react';
import styled, {keyframes, css} from 'styled-components';

export enum LoaderType {
  LIGHT = 'LIGHT',
  DARK = 'DARK',
}

type LoaderProps = {
  type?: LoaderType;
}

const move = keyframes`
  0% { transform: translateX(-100%); }
  100% { transform: translateX(100%); }
`;

const animation = css`
  animation: ${move} 1.5s linear infinite forwards;
`
const StyledLoaderWrapper = styled.div`
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 9999;
`;

const StyledLoader = styled.div<{ type: LoaderType }>`
  display: block;
  width: 100%;
  height: 7px;
  background: 
    ${({type, theme}) => (type === LoaderType.DARK ? theme.palette.PRIMARY : theme.palette.LIGHT_MIN)};
  ${animation};
`;

const Loader:React.FC<LoaderProps> = ({ type = LoaderType.DARK }) => (
  <StyledLoaderWrapper>
    <StyledLoader type={type}/>
  </StyledLoaderWrapper>
);

export default Loader;
