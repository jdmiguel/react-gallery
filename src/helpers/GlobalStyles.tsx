import { createGlobalStyle } from 'styled-components';

export const GlobalStyles = createGlobalStyle`
  body {
    margin: 0;
    padding: 0;
    font-family: 'Josefin Sans', sans-serif;
    font-size: 15px;
    color: #2f3235;
    background-color: #F3F3F3;
  }

  h1, h2, h3, h4, h5, h6, p{
    margin: 0;
  }

  #root{
    overflow: auto;
  }
`;