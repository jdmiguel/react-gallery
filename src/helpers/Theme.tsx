import React, { ReactNode } from 'react';
import { DefaultTheme, ThemeProvider } from 'styled-components';

export const theme: DefaultTheme = {
  palette: {
    PRIMARY: '#A66885',
    PRIMARY_DARK: '#734359',
    PRIMARY_LIGHT: '#F2C1AE',
    LIGHT_MAX: '#E5EAED',
    LIGHT_MIN: '#C8C8C8',
    GREY: '#E5EAED',
    DARK_MAX: '#2f3235',
    DARK_MIN: '#798186',
  },
};

export const renderWithTheme = (component: ReactNode) => (
  <ThemeProvider theme={theme}>{component}</ThemeProvider>
);
