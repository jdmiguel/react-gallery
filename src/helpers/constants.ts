// API
export const API_PIXABAY = "https://pixabay.com/api/";
export const API_KEY = "6473511-0417f2cad683f1bee54cafe15";

// IMAGES
export const DEFAULT_THUMBS = [
  {
    id: 0,
    src: '',
    title: ''
  },
];

// IMAGE LOADING
export const INITIAL_PAGE = 1;
export const MAX_PAGES_ALLOWED = 20;
export const INITIAL_MAX_IMAGES_LOADED = 40;
export const MAX_IMAGES_LOADED_BY_SCROLL = 20;
export const MIN_FOOTER_VIEWS = 3;
export const OBSERVER_OPTIONS = {
  root: document.querySelector('root'),
  rootMargin: '15%',
  threshold: 0,
};