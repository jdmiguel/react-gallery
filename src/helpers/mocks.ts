export const THUMB_PROPS_MOCKED = {
  id: 3063284,
  src: "https://pixabay.com/get/55e0d340485aa814f1dc84609629327f113dd7e05b4c704f752f78d79744c15e_640.jpg",
  title: "flower"
};
export const BUTTON_TEXT_MOCKED = "default button";
export const Pill_TEXT_MOCKED = "default pill";