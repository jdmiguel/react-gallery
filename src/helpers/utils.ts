export const getLoweredText = (text: string) => text.toLowerCase();