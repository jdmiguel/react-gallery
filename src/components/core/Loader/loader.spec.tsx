import React from 'react';
import { render } from '@testing-library/react';

import Loader, { LoaderType } from '.';

import { renderWithTheme } from '../../../helpers/Theme';

describe('Component: Loader', () => {
  it('should render dark version', () => {
    const { container } = render(
      renderWithTheme(<Loader />)
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render light version', () => {
    const { container } = render(
      renderWithTheme(<Loader type={LoaderType.LIGHT}/>)
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
