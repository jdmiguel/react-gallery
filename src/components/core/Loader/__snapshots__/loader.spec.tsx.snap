// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Loader should render dark version 1`] = `
.c0 {
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 9999;
}

.c1 {
  display: block;
  width: 100%;
  height: 7px;
  background: #A66885;
  -webkit-animation: jIfmAB 1.5s linear infinite forwards;
  animation: jIfmAB 1.5s linear infinite forwards;
}

<div
  class="c0"
>
  <div
    class="c1"
    type="DARK"
  />
</div>
`;

exports[`Component: Loader should render light version 1`] = `
.c0 {
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 9999;
}

.c1 {
  display: block;
  width: 100%;
  height: 7px;
  background: #C8C8C8;
  -webkit-animation: jIfmAB 1.5s linear infinite forwards;
  animation: jIfmAB 1.5s linear infinite forwards;
}

<div
  class="c0"
>
  <div
    class="c1"
    type="LIGHT"
  />
</div>
`;
